ts/modules/marker-clusters.ts

```diff
@@ -0,0 +1,1948 @@
+    },
+    /**
+     * Fires when the cluster point is clicked and `drillToCluster` is enabled.
+     * One parameter, `event`, is passed to the function. The default action
+     * is to zoom to the cluster points range. This can be prevented
+     * by calling `event.preventDefault()`.
+     *
+     * @sample {highcharts} highcharts/plotoptions/series-events-legenditemclick/
+     *         Confirm hiding and showing
+     *
+     * @type      {Highcharts.OnDrillToClusterCallbackFunction}
+     * @product   highcharts highmaps
+     * @see [cluster.drillToCluster](#plotOptions.scatter.marker.cluster.drillToCluster)
+     * @optionparent plotOptions.scatter.marker.cluster.events.onDrillToCluster
+     */
+
+    /**
+     * An array defining zones within marker clusters.
+     *
+     * In styled mode, the color zones are styled with the
+     * `.highcharts-cluster-zone-{n}` class, or custom
+     * classed from the `className`
+     * option.
+     *
+     * @sample highcharts/marker-clusters/basic
+     *         Marker clusters zones
+     * @sample maps/marker-clusters/custom-alg
+     *         Zones on maps
+     *
+     * @type      {Array<ClusterZonesOptions>}
+     * @product   highcharts highmaps
+     * @optionparent plotOptions.scatter.marker.cluster.zones
+     */
+
+    /**
+     * Styled mode only. A custom class name for the zone.
+     *
+     * @sample highcharts/css/color-zones/
+     *         Zones styled by class name
+     *
+     * @type      {string}
+     * @optionparent plotOptions.scatter.marker.cluster.zones.className
+     */
+
+    /**
+     * Styles for the cluster zone.
+     *
+     * @see [StyleOptions](#plotOptions.scatter.marker.cluster.style)
+     * @extends   plotOptions.scatter.marker.cluster.style
+     * @type      {StyleOptions}
+     *
+     * @product   highcharts highmaps
+     * @optionparent plotOptions.scatter.marker.cluster.zones.style
+     */
+
+    /**
+     * The value where the zone starts.
+     *
+     * @type      {number}
+     *
+     * @product   highcharts highmaps
+     * @optionparent plotOptions.scatter.marker.cluster.zones.from
+     */
+
+    /**
+     * The value where the zone ends.
+     *
+     * @type      {number}
+     *
+     * @product   highcharts highmaps
+     * @optionparent plotOptions.scatter.marker.cluster.zones.to
+     */
+
+    /**
+     * The fill color of the cluster marker in hover state. When
+     * `undefined`, the series' or point's fillColor for normal
+     * state is used.
+     *
+     * @type      {Highcharts.ColorType}
+     * @optionparent plotOptions.scatter.marker.cluster.states.hover.fillColor
+     *
+     */
+
+    /**
+     * Options for the cluster data labels.
+     * @extends   plotOptions.series.dataLabels
+     * @type    {Highcharts.DataLabelsOptionsObject}
+     */
+    dataLabels: {
+        enabled: true,
+        format: '{point.clusterPointsAmount}',
+        verticalAlign: 'middle',
+        align: 'center'
+    }
```

Comment:
Should be have a `@default` tag with a JSON of the sub properties. The sub properties should be `@ignore-option` because they are part of the `Highcharts.DataLabelsOptionsObject` interface. Therefore:
```ts
      /**
       * Options for the cluster data labels.
       * @type    {Highcharts.DataLabelsOptionsObject}
       * @default { "enabled": true, "format": "{point.clusterPointsAmoun}", "verticalAlign": "middle", "align": "center" }
       */
      dataLabels: {
         /** @ignore-option */
         enabled: true,
         /** @ignore-option */
         format: '{point.clusterPointsAmount}',
         /** @ignore-option */
         verticalAlign: 'middle',
         /** @ignore-option */
         align: 'center'
      }
```